modules/mod_auth_file.c

```diff
@@ -632,15 +628,15 @@ static int af_setgrent(pool *p) {
       return -1;
     }
 
-    /* As the file may contain sensitive data, we do not want it lingering
-     * around in stdio buffers.
-     */
-    (void) setvbuf(af_group_file->af_file, NULL, _IONBF, 0);
```

Comment:
Actually, I wonder if it is this `setvbuf(3)` call which is causing `fgets(3)` (and other library `FILE *` functions) to do the non-buffered reads.  Quite possibly.

Still, I think this is a good change, as it removes more "moving parts", and ensures that what we are doing is more within the ProFTPD source code.